### PR TITLE
LOG: New RTSP log module (aka "Improve logs of the HTTP module v2")

### PR DESCRIPTION
**Pull request by lars18th**
The RTSP log is separated from the HTTP log.

Supersedes #726 .

### CODE REVIEW

**Comment by lars18th** [comment]
Hi @catalinii ,

I'm very satisfied with the results. Thank you for your tips!
Please, merge it. :wink:

**Comment by lars18th** [comment]
Hi @catalinii ,

You agree with this implementation? I feel that it's clean and that it meets your requirements.

Please, merge it or comment about it.
Thank you!

File: src/minisatip.c [hunk 1]
```diff
@@ -1908,7 +1920,11 @@ void http_response(sockets *s, int rc, char *ah, char *desc, int cseq, int lr)
 		(lresp == sizeof(resp) - 1) ? "(message truncated) " : "", s->sock,
 		get_sockaddr_host(s->sa, ra, sizeof(ra)), get_sockaddr_port(s->sa),
 		lr, s->id);
-	LOGM("MSG client << process :\n%s", resp);
+	//LOGM("MSG client << process :\n%s", resp);
```
Comment (catalinii) [inline src/minisatip.c:1923]: remove this

Comment (catalinii) [reply src/minisatip.c:1923]: You have multiple log lines in read_rtsp, do you mind updating all of them ?

Comment (lars18th) [reply src/minisatip.c:1923]: > remove this

This is not an error. I leave here to understand the next block. However, if you want I'll remove it.

Comment (lars18th) [reply src/minisatip.c:1923]: > You have multiple log lines in read_rtsp, do you mind updating all of them ?

You mean change from LOG to LOGM in the `read_rtsp()` function?

**Comment by lars18th** [comment]
Hi @catalinii ,

All done and ready to merge it! :wink:

**Comment by catalinii** [comment]
I meant that all the LOG and LOGM lines should be changed:

1081:		LOG(
1106:			LOG(
1111:		LOG(
1127:	LOG("Read RTSP (handle %d) [%s:%d] sid %d, len: %d, sock %d", s->sid,
1130:	LOGM("MSG client >> process :\n%s", s->buf);
1141:		LOG_AND_RETURN(0,
1165:		LOGM("Updating sid %d time to %jd, current time %jd", sid->sid, s->rtime, getTick());
1354:			LOG(
1418:	LOG("Read HTTP (handle %d) [%s:%d] sid %d, sock %d", s->sid,
1421:	LOGM("MSG client >> process :\n%s", s->buf);
1424:	//      LOG("args: %s -> %s -> %s",arg[0],arg[1],arg[2]);
1527:	LOG("Requested sid close %d timeout %d type %d", s->sid,



Also please make sure you create just one commit (instead of multiple). Is easier to trakc the history of what went in.

**Comment by lars18th** [comment]
Hi,

> Also please make sure you create just one commit (instead of multiple). Is easier to trakc the history of what went in.

I know it, and every time it's possible I'll do. :wink:

**Comment by lars18th** [comment]
Hi @catalinii 

> I meant that all the LOG and LOGM lines should be changed:

I what sense? In the list you provided there are different cases and _all_ IMHO are right. Note this:
- Nearly all cases you listed are **NOT** shared between rtsp and http. So, it's not required to add any absurd code to check the log module. The well added `#undef DEFAULT_LOG     #define DEFAULT_LOG LOG_XXX` lines provide the correct, simple and elegant solution.
- One or two cases that you listed are already fixed. Please, check the code in deep.

Futhermore, I need to comment:
1. The only case where rtsp and http share identical logging code is inside the function `http_response()`. So, **only** in line 1911 is required a change (that is already in the patch).
2. I never push a PR before testing it. I can confirm that all lines using `-l general` and/or `-l http` and `-l rtsp` are absolutly correct when running minisatip. I do a lot of tests using RTSP and/or HTTP clients and activating different log levels and modules. In all tests the printed lines are the expected.
3. And with all due respect, I encourage you to test the patch.

Regards.

**Comment by lars18th** [comment]
Hi @catalinii ,

Waiting for the merge of this PR.

**Comment by lars18th** [comment]
Hi @catalinii ,

I'm preparing a new PR with a lot of new enhancements for the logging functionality. Please stay on hold until I publish it.

**Comment by catalinii** [comment]
Please create just one commit.

**Comment by lars18th** [comment]
Hi @catalinii ,

I withdraw this patch as it has been supersed by #744 .

**Pull request closed without merging**